### PR TITLE
Bug 2038250: Fix catalogsource name

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
The CI test `e2e-metal-assisted-operator-ztp-periodic-upgrade` fails on `release-ocm-2.4` branch because of the missing fix openshift/assisted-service@22868a6. 
This PR adds the fix from the master branch to this branch.

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [x] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: deploy/operator/setup_assisted_operator.sh [hunk 1]
```diff
@@ -96,7 +96,23 @@ EOF
 
 function install_from_catalog_source() {
   catalog_source_name="${1}"
+  if [ "${ASSISTED_UPGRADE_OPERATOR}" = "true" ]; then
+   catalog_source=${ASSISTED_SERVICE_OPERATOR_CATALOG}
```
Comment (osherdp) [inline deploy/operator/setup_assisted_operator.sh:100]: nit: 
```suggestion
    catalog_source=${ASSISTED_SERVICE_OPERATOR_CATALOG}
```

**Review by osherdp — APPROVED** [review]

**Comment by mkowalski** [comment]
/test e2e-metal-assisted-kube-api-late-binding-single-node

Timeout on `ContainerCreating`, may be timeout on quay.io or something

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Comment by pawanpinjarkar** [comment]
/bugzilla/valid-bug

**Comment by osherdp** [comment]
/hold
I think there's a real issue here:
```
time="2021-12-22T19:02:52Z" level=fatal msg="OpenShift versions is empty" func=main.main file="/go/src/github.com/openshift/origin/cmd/main.go:205"
```

**Comment by osherdp** [comment]
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node
This job is probably totally broken

**Review by osherdp — APPROVED** [review]

**Pull request merged**